src/Illuminate/Foundation/Console/TinkerCommand.php

```diff
@@ -26,11 +40,64 @@ class TinkerCommand extends Command {
 	 */
 	public function fire()
 	{
-		$this->getApplication()->setCatchExceptions(false);
+		$app = $this->getApplication();
+		$app->setCatchExceptions(false);
 
-		$shell = new Shell();
+		$config = new Configuration();
+
+		$pm = $config->getPresenterManager();
+		$pm->addPresenters($this->getPresenters());
+
+		$shell = new Shell($config);
+		$shell->addCommands($this->getCommands());
+		$shell->setIncludes($this->argument('include'));
+		$shell->setScopeVariables([
```

Comment:
Or you could tell me what other variables should be added to the scope, and we could leave it multi-line ;)